core/types/subtyping.cc

```diff
@@ -1383,9 +1383,21 @@ bool isSubTypeUnderConstraintSingle(const GlobalState &gs, TypeConstraint &const
                                 break;
                             }
                         }
-                        auto message = ErrorColors::format("`{}` is not {} `{}` for {} type member `{}`", a1i.show(gs),
-                                                           joiningText, a2j.show(gs), variance, idxTypeMember.show(gs));
-                        subCollector.message = message;
+                        // For proc types, make type member references more user-friendly
+                        if (a2->klass.data(gs)->name.show(gs).compare("Proc1") == 0) {
+                            auto message = ErrorColors::format("`{}` is not {} `{}`", a1i.show(gs), joiningText, a2j.show(gs));
+                            if (variance == "covariant") {
```

Comment:
Using the `variance` to determine whether it was an argument or a return is technically sound but indirect. Let's instead compare whether the `idxTypeMember->name` is `core::Names::Constants::Return`, because this is the name used for all the return types of these classes declared in `rbi/procs.rbi`.

e.g.

```cpp
if (idxTypeMember.data(gs)->name == core::Names::Constants::Return()) {
  // return
} else {
  // arg
}
```